### PR TITLE
CSI-2854 sync controller image when not running

### PR DESCRIPTION
**Pull request by oriyarde**
workaround for https://github.com/kubernetes/kubernetes/issues/67250

### CODE REVIEW

File: pkg/controller/ibmblockcsi/ibmblockcsi_controller.go [hunk 1]
```diff
@@ -338,16 +339,31 @@ func (r *ReconcileIBMBlockCSI) updateStatus(instance *ibmblockcsi.IBMBlockCSI, o
...
+	instance.Status.ControllerReady = controllerStatefulset.Status.ReadyReplicas == controllerStatefulset.Status.Replicas
 	instance.Status.NodeReady = node.Status.DesiredNumberScheduled == node.Status.NumberAvailable
 	phase := csiv1.DriverPhaseNone
 	if instance.Status.ControllerReady && instance.Status.NodeReady {
 		phase = csiv1.DriverPhaseRunning
 	} else {
-		if originalStatus.ControllerReady && !instance.Status.ControllerReady {
-			controllerRestart = true
+		if !instance.Status.ControllerReady {
+			err := r.getControllerPod(controllerStatefulset, controllerPod)
```
Comment (ArbelNathan) [inline pkg/controller/ibmblockcsi/ibmblockcsi_controller.go:349]: what is this for?

Comment (oriyarde) [reply pkg/controller/ibmblockcsi/ibmblockcsi_controller.go:349]: it fills the `controllerPod` instance with (non-empty) values

Comment (zingero) [reply pkg/controller/ibmblockcsi/ibmblockcsi_controller.go:349]: Don't we want to do that if the controller is ready as well?

File: pkg/controller/ibmblockcsi/ibmblockcsi_controller.go [hunk 2]
```diff
@@ -384,20 +400,37 @@ func (r *ReconcileIBMBlockCSI) updateStatus(instance *ibmblockcsi.IBMBlockCSI, o
...
-	statefulSetName := oconfig.GetNameForResource(oconfig.CSIController, name)
-	controllerPodName := fmt.Sprintf("%s-0", statefulSetName)
-	err := r.client.Get(context.TODO(), types.NamespacedName{
-		Name:      controllerPodName,
-		Namespace: namespace,
-	}, pod)
-
-	if err != nil {
-		return err
+func (r *ReconcileIBMBlockCSI) isAllPodImagesSynced(controllerStatefulset *appsv1.StatefulSet, controllerPod *corev1.Pod) bool {
```
Comment (zingero) [inline pkg/controller/ibmblockcsi/ibmblockcsi_controller.go:403]: I suggest renaming the function to `areAllPodImagesSynced`

File: pkg/controller/ibmblockcsi/ibmblockcsi_controller.go (hunk 1 above)
Comment (oriyarde) [reply pkg/controller/ibmblockcsi/ibmblockcsi_controller.go:349]: don't see why

File: pkg/controller/ibmblockcsi/ibmblockcsi_controller.go (hunk 2 above)
Comment (oriyarde) [reply pkg/controller/ibmblockcsi/ibmblockcsi_controller.go]: done

**Review by zingero — APPROVED** [review]

**Review by ArbelNathan — APPROVED** [review]
LGTM

**Pull request merged**